src/handler.ts

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IContentsModel, IContentsManager
+} from 'jupyter-js-services';
+
+import * as arrays
+  from 'phosphor-arrays';
+
+import {
+  IMessageFilter, IMessageHandler, Message, installMessageFilter,
+  removeMessageFilter
+} from 'phosphor-messaging';
+
+import {
+  IChangedArgs, Property
+} from 'phosphor-properties';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  Widget, Title
+} from 'phosphor-widget';
+
+import {
+  JupyterCodeMirrorWidget
+} from './widget';
+
+import {
+  loadModeByFileName
+} from './utils';
+
+
+/**
+ * An abstract base class for implementing file handlers.
+ */
+export
+abstract class AbstractFileHandler implements IMessageFilter {
```

Comment:
Codemirror mode handling is not done in this abstract class.  What I'm saying is that the abstract base class should be left where it was, and this module should import and inherit from it.